batcher/aligned-sdk/src/communication/batch.rs

```diff
@@ -0,0 +1,137 @@
+use std::{collections::HashSet, time::Duration};
+
+use futures_util::StreamExt;
+use log::{debug, error};
+use tokio::time::timeout;
+
+use crate::{
+    core::errors,
+    core::types::{
+        AlignedVerificationData, BatchInclusionData, VerificationCommitmentBatch,
+        VerificationDataCommitment,
+    },
+    eth::BatchVerifiedEventStream,
+};
+
+const AWAIT_BATCH_VERIFICATION_TIMEOUT: u64 = 60;
+
+pub async fn handle_batch_inclusion_data<'s>(
+    batch_inclusion_data: BatchInclusionData,
+    aligned_verification_data: &mut Vec<AlignedVerificationData>,
+    verification_data_commitments_rev: &mut Vec<VerificationDataCommitment>,
+    stream: &mut BatchVerifiedEventStream<'s>,
+    verified_batch_merkle_roots: &mut HashSet<Vec<u8>>,
+) -> Result<(), errors::SubmitError> {
+    handle_batch_inclusion_data_without_await(
+        batch_inclusion_data.clone(),
+        aligned_verification_data,
+        verification_data_commitments_rev,
+    )
+    .await?;
+
+    let batch_merkle_root = batch_inclusion_data.batch_merkle_root.to_vec();
+
+    if !verified_batch_merkle_roots.contains(&batch_merkle_root) {
+        await_batch_verification(stream, &batch_inclusion_data.batch_merkle_root).await?;
+        verified_batch_merkle_roots.insert(batch_merkle_root);
+    }
+
+    Ok(())
+}
+
+pub async fn handle_batch_inclusion_data_without_await<'s>(
+    batch_inclusion_data: BatchInclusionData,
+    aligned_verification_data: &mut Vec<AlignedVerificationData>,
+    verification_data_commitments_rev: &mut Vec<VerificationDataCommitment>,
+) -> Result<(), errors::SubmitError> {
+    debug!("Received response from batcher");
+    debug!(
+        "Batch merkle root: {}",
+        hex::encode(batch_inclusion_data.batch_merkle_root)
+    );
+    debug!("Index in batch: {}", batch_inclusion_data.index_in_batch);
+
+    let verification_data_commitment =
+        verification_data_commitments_rev.pop().ok_or_else(|| {
+            errors::SubmitError::GenericError("Verification data commitments are empty".to_string())
+        })?;
+
+    if verify_response(&verification_data_commitment, &batch_inclusion_data) {
+        aligned_verification_data.push(AlignedVerificationData::new(
+            &verification_data_commitment,
+            &batch_inclusion_data,
+        ));
+    }
+
+    Ok(())
+}
```

Comment:
everything in this function is synchronous, so no need to declare it async.

Reply:
Solved in [4c5f0a4](https://github.com/yetanotherco/aligned_layer/pull/583/commits/4c5f0a4b4908438a42f276a92094b43715827b6a).